code/processes/gateway.q

```diff
@@ -330,7 +331,7 @@ getserversindependent:{[req;att;besteffort]
  alldone:1+first where all each all each' maxs value s;
 
  if[(null alldone) and not besteffort;
-        '"getserversindependent: cannot satisfy query as not all attributes can be matched"];
+  .gw.formatresponse[0b;`sync;"getserversindependent: cannot satisfy query as not all attributes can be matched"]];
```

Comment:
I don't think you need the response formatter on any of these getservers* functions.  These are invoked in both asyncexec and syncexec. 

If you look at line 437 the getservers function is invoked in an error trap and then you have added a format function to the result of that.  So I think the way you have it currently it would get passed through the formatter twice.  

It is also invoked in syncexecj at line 458.  I think what you need to do is change the call at 458 to error trap the result, and then pass it through the result formatter e.g. something like: 

serverids:@[getserverids;servertype;{.gw.formatresponse[0b;`sync;.gw.errorprefix,x]}];